### PR TITLE
improvement: add a skaffold profile to only build and publish the artifacts

### PR DESCRIPTION
**Pull request by mike-ensor**
Allow the release profile to build containers that are not development using skaffold sha256 build module

**[EDIT]**

### Description
- Add a section to describe how to just generate the manifests based on the source code locally available

image

### CODE REVIEW

**Comment by mike-ensor** [comment]
@Shabirmean -- I cannot see why this is failing in CB, can you take a look?

File: skaffold.yaml [hunk 1]
```diff
@@ -74,7 +75,23 @@ profiles:
       - k8-manifests/dev/api-server.yaml
       - k8-manifests/dev/inventory.yaml
       - k8-manifests/dev/payments.yaml
+- name: publish
```
Comment (bourgeoisor) [inline skaffold.yaml:78]: Is it a common use-case to use Skaffold modules like this? (e.g. a different module for "deploy" and for "build" steps).
Why not put both the build and deploy steps in the same profile and use different built-in flags for `skaffold` depending on if you just want to build, or if you want to build+deploy?

Comment (bourgeoisor) [reply skaffold.yaml:78]: Here's an example with Bank of Anthos: https://github.com/GoogleCloudPlatform/bank-of-anthos/blob/main/skaffold.yaml#L47-L77

Comment (Shabirmean) [reply skaffold.yaml:78]: We also have the same pattern (as in BoA) in this repo. Our use case was that we wanted to only build the images and not publish them anywhere. I see that using `build` command does exactly that. I think we must have missed it. 

Let me check again on this and get back!

Comment (Shabirmean) [reply skaffold.yaml:78]: I checked locally, and when I do: `skaffold build -p dev --default-repo=<AR_REPO/GCR_REPO> -t <TAG>`

it seems like the local changes are built and published to the repository _(with tagging)_ and **not deployed to the cluster. @mike-ensor can we use it for your use case? Let us know.

Comment (Shabirmean) [reply skaffold.yaml]: @bourgeoisor - reverted the changes and added a guide as to how to only build without deploy. Thank you for raising this.

**Review by bourgeoisor — DISMISSED** [review]
LGTM!

**Comment by Shabirmean** [comment]
:zap:  Two deployments have been created for the Point-of-Sale application. You may access and test them at: :zap: </br>
- **MySQL DB** backed deployment: [**34.132.217.104**](http://34.132.217.104)
- **Embedded H2** DB backed deployment: [**34.136.167.179**](http://34.136.167.179)

**Pull request merged**